server/src/modules/auth/password-revalidate.guard.ts

```diff
@@ -0,0 +1,29 @@
+import { Injectable, CanActivate, ExecutionContext, Inject, forwardRef } from '@nestjs/common';
+import { UsersService } from '@services/users.service'
+import { Observable } from 'rxjs';
+const bcrypt = require('bcrypt')
+
+@Injectable()
+export class PasswordRevalidateGuard implements CanActivate {
+
+  constructor(
+    private usersService: UsersService
+  ) {}
+
+  async validateUser(email: string, password: string): Promise<boolean> {
```

Comment:
Can we re-use the `validateUser` method from auth service?

Reply:
I considered this, however, the `validateUser` method comes from `AuthModule`, which references `UserModule`, creating a circular dependency. There are ways to circumvent this, however I felt it'd be less complicated to define another `validateUser` method, plus, the method I defined is slightly customized for the requirement at hand.